ompi/mca/mtl/ofi/mtl_ofi_component.c

```diff
@@ -197,6 +197,20 @@ ompi_mtl_ofi_component_init(bool enable_progress_threads,
     prov = providers;
 
     /**
+     * Skip the 'sockets' provider by default unless it has been
+     * specifically selected via the command line option.
+     *
+     * e.g. --mca mtl_ofi_provider sockets
+     */
+    if (0 == (strncmp(prov->fabric_attr->prov_name, "sockets", 7)) &&
```

Comment:
It might be better to do something like what has been done in other areas of OMPI:

* Have `mtl_ofi_prov_include` and `mtl_ofi_prov_exclude` MCA vars (change to `pml_ofi_` whenever relevant).
* Each MCA var can be a comma-delimited list of names.
* If `_include` is specified, use only the providers that are listed in the value (and error if any of them can't be used).
* If `_exclude` is specified, *exclude* the providers are are listed, and use any other available providers.
* If both `_include` and `_exclude` are specified (*by the user*), error.
* The OFI MTL can set the default value of `_include` to NULL, and the default value of `_exclude` to `sockets`

This way, the only hard-coding of "exclude the sockets provider" is in an MCA var default value, not in any core selection logic.  And if a user specifies either an `_include` or `_exclude` value, it overrides the default `_exclude` value.